controller/stack_resources.go

```
@@ -318,3 +319,55 @@ func (c *StackSetController) ReconcileStackRouteGroup(ctx context.Context, stack
 		routegroup.Name)
 	return nil
 }
+
+// Given the definition of ConfigurationResources on the StackTemplate, the
+// Stackset Controller will update the named user-provided ConfigMap to be
+// attached to the Stack by ownerReferences.
+//
+// User update of versioned ConfigMaps is not encouraged but is allowed for
+// consideration of emergency needs.
+func (c *StackSetController) ReconcileStackConfigMap(
+	ctx context.Context,
+	stack *zv1.Stack,
+	existing []*apiv1.ConfigMap,
+	generateUpdated func(*apiv1.ConfigMap) (*apiv1.ConfigMap, error),
+) error {
+	if stack.Spec.ConfigurationResources == nil {
+		return nil
+	}
+
+	if len(existing) >= len(stack.Spec.ConfigurationResources) {
+		return nil
+	}
+
+	for _, rsc := range stack.Spec.ConfigurationResources {
+		configMap, err := c.client.CoreV1().ConfigMaps(stack.Namespace).Get(ctx, rsc.ConfigMapRef.Name, metav1.GetOptions{})
```

If `rsc.ConfigMapRef.Name` doesn't follow the pattern `"{{{APPLICATION}}}-{{{VERSION}}}-configmap-name"` then we should return an error so that the Stack isn't created.